pkg/util/chanmanager/chanmanager.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Channel Manager keeps track of multiple channels
+package chanmanager
+
+import (
+	"fmt"
+	"sync"
+)
+
+type ChanManager interface {
+	// Creates a new channel with specified buffer size. Adds it to manager with the specified chanID.
+	// Returns a read-only version of the newly created channel.
+	CreateAndAddChan(chanId string, bufferSize uint) <-chan interface{}
+
+	// Closes and removes the channel associated with chanID.
+	// Returns an error if no associated channel exists.
+	CloseAndRemoveChan(chanId string) error
+
+	// Attempts to send msg to the channel associated with chanID.
+	// Returns an error if no associated channel exists.
+	SignalChan(chanId string, msg interface{}) error
```

Comment:
Perhaps Start, Close, and Send as the three method names?

also perhaps just 'id' instead of chanId (which should be chanID anyway)

Sorry for nitpicky naming comments, but I think it will make it easier for people to read this.